bigml/ensemble.php

```diff
@@ -132,6 +140,50 @@ public function __construct($ensemble, $api=null, $max_models=null, $storage="st
          $this->multi_model = new MultiModel($models, $this->api);
       }
 
+      if ($this->distributions == null) {
+          try {
+              $this->distributions = array();
+              foreach ($models as $model) {
+                  $distributions = $this->distributions;
+                  $distributions[] = array('training'=>array('categories'=> $model->tree->distribution));
+              }
+          }
+          catch(Exception $e) {
+              $this->distributions = array();
+              foreach ($models as $model) {
+                  $distributions = $this->distributions;
+                  $distributions[] = $model->object->model->distribution;
+              }
+          }
+      }
+
+      if (!$this->regression) {
```

Comment:
I think there might be a problem here. If the ensemble is boosted and you happen to recover the distribution from the individual models, although the ensemble has a categorical objective, the models will not and the exception part might fail... just check. In the python bindings versin there's an addition "non-boosted" check. We can discuss.